nomos-services/data-availability/dispersal/src/adapters/wallet/mod.rs

```diff
@@ -0,0 +1,21 @@
+pub mod mock;
+
+use nomos_core::{
+    da::BlobId,
+    mantle::{ops::channel::Ed25519PublicKey, SignedMantleTx},
+};
+
+#[async_trait::async_trait]
+pub trait DaWalletAdapter {
+    type Error;
+
+    // TODO: Pass relay when wallet service is defined.
+    fn new() -> Self;
+
+    fn blob_tx(
+        &self,
+        blob_id: BlobId,
+        blob_size: usize,
+        signer: Ed25519PublicKey,
+    ) -> Result<SignedMantleTx, Self::Error>;
+}
```

Comment:
I'm not sure this is gonna be the API. Maybe we should expect a fully formed `SignedMantleTx`, but for now it may do.

Reply:
We'll need some party to sign our blob TX, who ever that party will be this adapter will pass required blob data and form SignedMantleTx. For now it's called DaWalletAdapter, will rename when destination service is known.

Reply:
Dispersal service can't get SignedTx before encoding data. Just after encoding and getting the blob id we can form the TX